nixos/modules/services/web-apps/immich.nix

```diff
@@ -0,0 +1,247 @@
+{ config, lib, pkgs, self, ... }:
+let
+  cfg = config.services.immich;
+  isUnixSocket = lib.hasPrefix "/" cfg.database.host;
+  postgresEnv = if isUnixSocket then {
+    DB_URL = "socket://${cfg.database.host}?dbname=${cfg.database.name}";
+  } else {
+    DB_HOSTNAME = cfg.database.host;
+    DB_PORT = toString cfg.database.port;
+    DB_DATABASE_NAME = cfg.database.name;
+    DB_USERNAME = cfg.database.user;
+  };
+
+  commonServiceConfig = {
+    Type = "simple";
+    Restart = "on-failure";
+    RestartSec = 3;
+
+    # Hardening
+    CapabilityBoundingSet = "";
+    NoNewPrivileges = true;
+    PrivateUsers = true;
+    PrivateTmp = true;
+    PrivateDevices = true;
+    PrivateMounts = true;
+    ProtectClock = true;
+    ProtectControlGroups = true;
+    ProtectHome = true;
+    ProtectHostname = true;
+    ProtectKernelLogs = true;
+    ProtectKernelModules = true;
+    ProtectKernelTunables = true;
+    RestrictAddressFamilies = [
+      "AF_INET"
+      "AF_INET6"
+      "AF_UNIX"
+    ];
+    RestrictNamespaces = true;
+    RestrictRealtime = true;
+    RestrictSUIDSGID = true;
+  };
+  inherit (lib) types mkIf mkOption mkEnableOption;
+in {
+  options.services.immich = {
+    enable = mkEnableOption "Immich";
+    package = lib.mkPackageOption pkgs "immich" {};
```

Comment:
```suggestion
```
I doubt there's a need for this.

Reply:
++ for keeping it. I quite often make use of such capabilities.

Reply:
> why remove the flexibility?

I'm in favor of as few options as possible but it's fine if you really want it.